Removed usage of id in accordion css file

Rajdeepc opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

Removed usage of id in accordion css file
## Related issue(s)

<!---
    This project only accepts pull requests related to open issues

    - If suggesting a new feature or change, please discuss it in an issue first.
    - If fixing a bug, there should be an issue describing it with steps to reproduce.
-->

- #3059 

## Motivation and context

<!--- Why is this change required? What problem does it solve? -->

## Screenshots (if appropriate)

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply.  If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [ ] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

Westbrook commented:
Change looks great!

You asked how we version and release, so this is a good place to start that convo. We use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), I think the CSS project uses them, too, so hopefully nothing too new there. However, it's important to understand how those commits are used to mechanically create our [Changelog](https://github.com/adobe/spectrum-web-components/blob/main/CHANGELOG.md#0320-2023-06-01), the various "action types" in the conventional commit and consumed into the "Bug Fixes", "Features" and eventually "Breaking Changes" listings therein. In preparation for that relationship, it can be worth it to spend a moment thinking about what we want to communicate to our consumers with each change, or in some cases what we don't want to communicate.

In this change you're removed inert code that a consumer did not have access to leverage or change, as it would be delivered in the shadow root of the Accordion element. The question is do we _want_ to add the following to the Changelog for that:

```
### Bug Fixes

- removed usage of id in accordion ([4425c218](https://github.com/adobe/spectrum-web-components/pull/3290/commits/4425c218e9e10294751dfb4460ea6f5e45b7c37c))
```

I might lean towards not including it, as it's not something consumers need to actively know or do anything about, for which you'd want to then use a commit message of either `refactor: removed usage of id in accordion` or `chore: removed usage of id in accordion`, neither of which should be added to Changelog publication. However, I'm using this small change primarily as a way to share processes about the project and am happy to include it in the log if that is better for you.

Rajdeepc commented:
> Change looks great!
> 
> You asked how we version and release, so this is a good place to start that convo. We use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/), I think the CSS project uses them, too, so hopefully nothing too new there. However, it's important to understand how those commits are used to mechanically create our [Changelog](https://github.com/adobe/spectrum-web-components/blob/main/CHANGELOG.md#0320-2023-06-01), the various "action types" in the conventional commit and consumed into the "Bug Fixes", "Features" and eventually "Breaking Changes" listings therein. In preparation for that relationship, it can be worth it to spend a moment thinking about what we want to communicate to our consumers with each change, or in some cases what we don't want to communicate.
> 
> In this change you're removed inert code that a consumer did not have access to leverage or change, as it would be delivered in the shadow root of the Accordion element. The question is do we _want_ to add the following to the Changelog for that:
> 
> ```
> ### Bug Fixes
> 
> - removed usage of id in accordion ([4425c218](https://github.com/adobe/spectrum-web-components/pull/3290/commits/4425c218e9e10294751dfb4460ea6f5e45b7c37c))
> ```
> 
> I might lean towards not including it, as it's not something consumers need to actively know or do anything about, for which you'd want to then use a commit message of either `refactor: removed usage of id in accordion` or `chore: removed usage of id in accordion`, neither of which should be added to Changelog publication. However, I'm using this small change primarily as a way to share processes about the project and am happy to include it in the log if that is better for you.

Really appreciate if you could add this to the log! Also I will try to contribute my findings and observations too.

Westbrook reviewed (approved):
LGTM! Thanks for you first contribution 👏🏼 👏🏼 👏🏼 👏🏼 

Because it is in the commit, it will be in the log when this gets published.
